File: Practica1/Ejercicio4/src/practica1ejercicio4/AdminFiltros.java
package practica1ejercicio4;

public class AdminFiltros {
    private CadenaFiltros cadenaFiltros;
    
    public AdminFiltros(Objetivo correoFiltro){
        cadenaFiltros = new CadenaFiltros();
        cadenaFiltros.setObjetivo(correoFiltro);
    }
    
    public void addFiltro(Filtro filtro){
        cadenaFiltros.addFiltro(filtro);
    }
    
    public void postCorreo(Correo correo){
        cadenaFiltros.ejecutar(correo);
    }
}

File: Practica2/ejercicio_grupal1/lib/AdminFiltros.dart
import 'CadenaFiltros.dart';
import 'Objetivo.dart';
import 'Filtro.dart';
import 'Correo.dart';


// TODO: Añadir un filtro que sea para controlar si ya está ingresado uno o no
// TODO: Añadir una caja de correos registrados que se pueda acceder desde el main
// TODO: Mostrar los correos con su constraseña en obscure para ver cuales ya están

class AdminFiltros {
  final CadenaFiltros cadenaFiltros;

  AdminFiltros(Objetivo correoFiltro)
      : cadenaFiltros = CadenaFiltros() {
    cadenaFiltros.setObjetivo(correoFiltro);
  }

  void addFiltro(Filtro filtro) {
    cadenaFiltros.addFiltro(filtro);
  }

  MapEntry<bool,String> postCorreo(Correo correo) {
    return cadenaFiltros.ejecutar(correo);
  }
}